versioned_docs/version-0.82/examples/cruddy.mdx

```diff
@@ -0,0 +1,2482 @@
+            // query string is anything else or empty e.g. localhost:8080
+            _ => "Anonymous".to_string(),
+        };
+        response_body // [!code --]
+            .write() // [!code --]
+            .unwrap() // [!code --]
+            .blocking_write_and_flush(b"Hello from Rust!\n") // [!code --]
+            .unwrap(); // [!code --]
+        OutgoingBody::finish(response_body, None).expect("failed to finish response body"); // [!code --]
+        ResponseOutparam::set(response_out, Ok(response)); // [!code --]
+```
+
+This is the first time we're actually implementing a new WASI method in this component, so we should pause to consider exactly what we're doing:
+
+* When we look at our WIT bindings, we can see that `IncomingRequest` has a method `path_with_query()`. The original WIT tell us that `path_with_query()` returns a string of the request path, including any query. However, the string is contained within a return object and needs to be "unwrapped."
+* To get at the actual value, we use the `unwrap()` method.
+* To isolate the name, we split the value at the `=` and collect it as a `<Vec<&str>>` that we can use elsewhere.
+
+In order to guide our CRUD operations for the guestbook, we'd also like to know the method of the incoming request. We can detect that with the API as well. Try finding the method of the request with what you've learned so far.
+
+If you guessed that you could use a method on `request`, well done! Below the previous line add:
+
+```rust
+// Detect method // [!code ++]
+let method = request.method(); // [!code ++]
+```
+
+This will return a method of the `Method` type, which we will be able to use directly.
+
+We've implemented our basic HTTP operations. Now let's turn to key-value storage.  
+
+## Adding CRUD 
+
+We can follow the same procedure we used for `wasi-http` to explore the wasi-keyvalue API and start adding CRUD operations.  
+
+In `/wit/deps/keyvalue/types.wit`, we see that collections of key-value pairs are referenced as `buckets` in the API:
+
+```
+	/// A bucket is a collection of key-value pairs. Each key-value pair is stored
+	/// as a entry in the bucket, and the bucket itself acts as a collection of all
+	/// these entries.
+	///
+	/// It is worth noting that the exact terminology for bucket in key-value stores
+	/// can very depending on the specific implementation. For example,
+	/// 1. Amazon DynamoDB calls a collection of key-value pairs a table
+	/// 2. Redis has hashes, sets, and sorted sets as different types of collections
+	/// 3. Cassandra calls a collection of key-value pairs a column family
+	/// 4. MongoDB calls a collection of key-value pairs a collection
+	/// 5. Riak calls a collection of key-value pairs a bucket
+	/// 6. Memcached calls a collection of key-value pairs a slab
+	/// 7. Azure Cosmos DB calls a collection of key-value pairs a container
+	///
+	/// In this interface, we use the term `bucket` to refer to a collection of key-value
```

Comment:
This code that's indented ends up showing on the preview, could we de-indent this?